processing/data_collection/gazette/spiders/mg_itauna.py

```diff
@@ -0,0 +1,36 @@
+import datetime as dt
+
+from dateparser import parse
+from scrapy import Request
+
+from gazette.items import Gazette
+from gazette.spiders.base import BaseGazetteSpider
+
+
+class MgItaunaSpider(BaseGazetteSpider):
+    TERRITORY_ID = "3133808"
+    name = "mg_itauna"
+    allowed_domains = ["itauna.mg.gov.br"]
+    start_urls = ["https://www.itauna.mg.gov.br/portal/diario-oficial/"]
+
+    def parse(self, response):
+        diarios = response.xpath("//div[@class='d_e_modelo_diario']")
+        for diario in diarios:
+            url = (
+                "https://www.itauna.mg.gov.br"
+                + diario.css("a")[1].extract().split('"')[1]
+            )
+            date = diario.xpath("div/span/text()").extract()[0].split()[0]
+            is_extra_edition = True if len(diario.css("i")) == 4 else False
+            yield Gazette(
+                date=parse(date, languages=["pt"]).date(),
+                file_urls=[url],
+                is_extra_edition=is_extra_edition,
+                territory_id=self.TERRITORY_ID,
+                power="executive_legislature",
+                scraped_at=dt.datetime.utcnow(),
+            )
+        url = "https://www.itauna.mg.gov.br/portal/diario-oficial/{i}/0/0/0/0/"
+        last_page = int(response.xpath("//select/option/text()")[-1].extract())
+        for i in range(1, 1 + last_page):
+            yield Request(url.format(i=i), callback=self.parse)
```

Comment:
Maybe we could separate the pagination from the actual parsing? This could avoid confusion due to the "recursiveness".

Like:

```python
...
def parse(self, response):
	url = "https://www.itauna.mg.gov.br/portal/diario-oficial/{i}/0/0/0/0/"
	for i in range(1, last_page + 1):
        yield Request(url.format(i=i), callback=self.parse_editions_page)

def parse_editions_page(self, response):
	diarios = response.xpath("//div[@class='d_e_modelo_diario']")
    for diario in diarios:
	...
...
```

Reply:
This doesn't introduce confusion. It is a very well known pattern in spiders. The solution you proposed will also introduce an extra request to page `1` that could be avoided if we do everything in `parse` method. As the spider is short, it won't be a problem.

Reply:
Then maybe adding a `yield from self.parse_editions_page(response)` at the end of the `parse` method and change the range to start at 2?

This addresses the unnecessary request maintaining separations of concerns.

I said it could create some confusion because the loop is creating new requests to the same method that creates those same requests. If the dupefilter wasn't in place this would be an infinite loop. And it would be better to address this in an explicit way than a implicit one.

Should have explained it a bit better.